app/Main.hs

```diff
@@ -229,10 +226,9 @@ main = do
     -- force line buffering (always, not just when stdout is connected to a tty),
     -- it's critical for successful parsing of NITTA's stdout in python scripts
     hSetBuffering stdout LineBuffering
-
-    toml <- case uarch of
+    conf <- case uarch of
         Nothing -> return Nothing
-        Just path -> Just . getToml <$> T.readFile path
+        Just path -> Just <$> parseConfig path
```

Comment:
Something like `parseConfig <$> path` should work?

Reply:
`parseConfig` is a `FilePath -> IO MicroarchitectureConf` type
`path` is a `FilePath` type
`<$>` is a `(a -> b) -> f a -> f b` type

The result must be type of `IO Maybe MicroarchitectureConf`, so `Just <$> parseConfig path`doesn't seem to be able to be simplified.